packages/@aws-cdk/aws-codebuild/README.md

```diff
@@ -119,3 +124,103 @@ To define CloudWatch event rules for build projects, use one of the `onXxx` meth
+
+```ts
+const project = new codebuild.Project(this, 'MyProject', {
+    secondarySources: [
+        new codebuild.CodeCommitSource({
+            identifier: 'source2',
+            repository: repo,
+        }),
+    ]
+    secondaryArtifacts: [
+        new codebuild.S3BucketBuildArtifacts({
+            identifier: 'artifact2',
+            bucket: bucket,
+            path: 'some/path',
+            name: 'file.zip',
+        }),
+    ],
+    // ...
+});
+```
+
+The `identifier` property is required for both secondary sources and artifacts.
+
+The contents of the secondary source will be available to the build under the directory
+specified by the `CODEBUILD_SRC_DIR_<identifier>` environment variable
+(so, `CODEBUILD_SRC_DIR_source2` in the above case).
+
+The secondary artifacts have their own section in the buildspec,
+under the regular `artifacts` one.
+Each secondary artifact has its own section,
+beginning with their identifier.
+
+So, a buildspec for the above Project could look something like this:
+
+```ts
+const project = new codebuild.Project(this, 'MyProject', {
+    // secondary sources and artifacts as above...
+    buildSpec: {
+        version: '0.2',
+        phases: {
+            build: {
+                commands: [
+                    'cd $CODEBUILD_SRC_DIR_source2',
+                    'touch output2.txt',
+                ],
+            }
+        },
+        artifacts: {
+            'secondary-artifacts': {
+                'artifact2': {
+                    'base-directory': '$CODEBUILD_SRC_DIR_source2',
+                    'files': [
+                        'output2.txt',
+                    ],
+                },
+            },
+        },
+    },
+});
+```
+
+#### Multiple inputs and outputs in CodePipeline
+
+Unfortunately, you cannot use neither `secondarySources` nor `secondaryArtifacts`
+on a Project used in CodePipeline.
+However, you can get the same effect by multiple inputs and outputs when creating your CodeBuild CodePipeline Action.
+So, to achieve a similar build to what was shown above,
+you could do something like the following in CodePipeline:
+
+```ts
+const sourceStage = pipeline.addStage('Source');
+const sourceAction1 = repository1.addToPipeline(sourceStage, 'Source1');
+const sourceAction2 = repository2.addToPipeline(sourceStage, 'Source2', {
+    outputArtifactName: 'source2',
+});
+
+const buildStage = pipeline.addStage('Build');
+const buildAction = project.addBuildToPipeline(buildStage, 'Build', {
+    inputArtifact: sourceAction1.outputArtifact,
+    outputArtifactName: 'artifact1', // for better buildspec readability - see below
+    additionalInputArtifacts: [
+        sourceAction2.outputArtifact, // this is where 'source2' comes from
+    ],
+    additionalOutputArtifactNames: [
+        'artifact2',
+    ],
+});
+// use buildAction.additionalOutputArtifacts as inputs to later stages...
+```
+
+**Note**: when a CodeBuild Action in a Pipeline has more than one input,
```

Comment:
Is this section trying to say that you need to list both `artifact1` and `artifact2` under `secondary-artifacts`? An example might help to solidify that. I'm trying hard to parse this text but I'm finding it hard.

Reply:
Yes. Explanation fail as well. Will add an example and re-word.